### PR TITLE
Add config file creation when file does not exist

### PR DESCRIPTION
**Pull request by NekoSelena**

### CODE REVIEW

File: cubic-server/configuration/ConfigHandler.cpp [hunk 1]
```diff
@@ -11,6 +13,11 @@ namespace Configuration
 
     void ConfigHandler::parse(const std::string &path) {
         YAML::Node config;
+        if (!std::filesystem::exists(path)) {
+            std::ofstream configFile(path);
+            configFile << "network:\n  ip: 0.0.0.0\n  port: 25565\n\ngeneral:\n  max_players: 20\n  motd: A Cubic Server" << std::endl;
```
Comment (huntears) [inline cubic-server/configuration/ConfigHandler.cpp:18]: Please put a constexpr variable in the file to be able to change it afterwards.
And also make it a multiline to make editing and reading easier.

**Review by huntears — CHANGES_REQUESTED** [review]

File: cubic-server/configuration/ConfigHandler.cpp (hunk 1 above)
Comment (NekoSelena) [reply cubic-server/configuration/ConfigHandler.cpp]: I pushed a fix, it it what I was supposed to do ? I am not really sure how to use constexpr

**Review by Trompettesib — COMMENTED** [review]
Not sure about utility of a constexpr here but overall seem good to me

File: cubic-server/configuration/ConfigHandler.cpp [hunk 2]
```diff
@@ -11,6 +13,20 @@ namespace Configuration
 
     void ConfigHandler::parse(const std::string &path) {
         YAML::Node config;
+        if (!std::filesystem::exists(path)) {
+            std::ofstream configFile(path);
+            constexpr std::string_view fileContent =
```
Comment (huntears) [inline cubic-server/configuration/ConfigHandler.cpp:18]: Please put it outside of the function in the `Configuration` namespace, and call it something like `defaultConfigContent`

**Review by huntears — CHANGES_REQUESTED** [review]

**Review by Trompettesib — DISMISSED** [review]
LGTM

**Review by Trompettesib — APPROVED** [review]
👌

**Review by Trompettesib — APPROVED** [review]
👌

**Review by huntears — APPROVED** [review]

**Pull request merged**